cmake/CMakeLists.txt

```diff
@@ -23,13 +23,11 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CXX_FLAGS)
   set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel." FORCE)
 endif(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CXX_FLAGS)
 
-foreach(STYLE_FILE style_angle.h style_atom.h style_body.h style_bond.h style_command.h style_compute.h style_dihedral.h style_dump.h
-             style_fix.h style_improper.h style_integrate.h style_kspace.h style_minimize.h style_nbin.h style_npair.h style_nstencil.h
-             style_ntopo.h style_pair.h style_reader.h style_region.h)
-  if(EXISTS ${LAMMPS_SOURCE_DIR}/${STYLE_FILE})
-    message(FATAL_ERROR "There is a ${STYLE_FILE} in ${LAMMPS_SOURCE_DIR}, please clean up the source directory first")
-  endif()
-endforeach()
+# remove any style headers in the src dir
+file(GLOB SRC_STYLE_FILES ${LAMMPS_SOURCE_DIR}/style_*.h)
+if(SRC_STYLE_FILES)
+  file(REMOVE ${SRC_STYLE_FILES})
```

Comment:
No, it's "enforcing" coding standards. Creating a file with such a name is asking for it. I would simply say `style_` is reserved for the build system. So all it does is nuking files that shouldn't start with `style_` in the first place.